### PR TITLE
test(app-testing): add battery of protocols to analyze in the app

### PR DESCRIPTION
**Pull request by y3rsh**
# Overview

https://user-images.githubusercontent.com/502770/210871575-5ff95c6b-1bf0-4f34-9bf7-571ac9ccda2f.mp4

- Perform app analysis on a variety of protocols to validate a build.  Runs on Windows, Mac, and Linux
- Switch to ruff, no flake8 or isort

## Risk assessment

None, test only

### CODE REVIEW

File: app-testing/conftest.py [hunk 1]
```diff
@@ -3,17 +3,16 @@
...
 from rich import pretty, traceback
 from rich.console import Console
 from selenium import webdriver
 from selenium.webdriver.chrome.options import Options
 from selenium.webdriver.chrome.webdriver import WebDriver
 
-from automation.menus.left_menu import LeftMenu
-from automation.pages.app_settings import AppSettings
-from automation.resources.robot_data import ROBOT_MAPPING, RobotDataType
-
```
Comment (y3rsh) [inline app-testing/conftest.py:16]: I prefer the 3rd party separate, maybe can change with a setting in ruff?  Not going to change now unless there are strong feelings.

File: .github/workflows/app-testing-lint.yaml [hunk 2]
```diff
@@ -29,11 +29,9 @@ jobs:
...
           python-version: '3.11'
           cache: 'pipenv'
-          cache-dependency-path: |
-            app-testing/Pipfile.lock
+          cache-dependency-path: app-testing/Pipfile.lock
       - name: Install Pipenv
-        run: |
-          pip install -U pipenv
+        run: pip install -U pipenv
       - name: Pipenv Install
```
Comment (y3rsh) [inline .github/workflows/app-testing-lint.yaml:35]: This name is not clear. Fix.

File: app-testing/locators.py [hunk 3]
```diff
@@ -80,25 +77,19 @@ def instantiate(driver: WebDriver, console: Console) -> None:
     global dev
     dev = src.resources.ot_robot.OtRobot(console, src.resources.robot_data.Dev())
     global emulated_alpha
-    dev = src.resources.ot_robot.OtRobot(
-        console, src.resources.robot_data.EmulatedAlpha()
-    )
+    dev = src.resources.ot_robot.OtRobot(console, src.resources.robot_data.EmulatedAlpha())
```
Comment (DerekMaggio) [inline app-testing/locators.py]: Should this variable be `emulated_alpha` 
I don't see you setting its value besides when you set it to `None`.

File: app-testing/automation/data/protocols.py [hunk 4]
```diff
@@ -26,3 +27,117 @@ class Protocols:
...
+
+    OT2_P300S_Thermocycler_Moam_Error: Protocol = Protocol(
+        file_name="OT2_P300S_Thermocycler_Moam_Error",
+        file_extension="py",
+        protocol_name="OT2_P300S_Thermocycler_Moam_Error.py",
+        robot="OT-2",
+        app_error=True,
+        robot_error=True,
+        app_analysis_error="DeckConflictError [line 19]: thermocyclerModuleV2 in slot 7 prevents thermocyclerModuleV1 from using slot 7.",  # noqa: E501
+        robot_analysis_error="?",
```
Comment (DerekMaggio) [inline app-testing/automation/data/protocols.py:57]: What does it mean when `robot_analysis_error="?"`
I might of missed it, but I didn't see any handling for this in your code

Comment (y3rsh) [reply app-testing/automation/data/protocols.py:57]: 💯 placeholder.  If I use the protocol in the DB snapshot script, this field will get populated.

File: app-testing/locators.py (hunk 3 above)
Comment (y3rsh) [reply app-testing/locators.py]: Yes, it should be `emulated_alpha` 
Never caught it because it was a placeholder.

Comment (y3rsh) [inline app-testing/locators.py]:
```suggestion
    emulated_alpha = src.resources.ot_robot.OtRobot(console, src.resources.robot_data.EmulatedAlpha())
```

**Review by nusrat813 — APPROVED** [review]
LGTM

**Review by DerekMaggio — APPROVED** [review]

**Pull request merged**